oops/backplane/unittester_support.py

```diff
@@ -0,0 +1,503 @@
+            sys.exit()
+
+        if oldrec.startswith('**'):
+            first_test_results_found = True
+            continue
+
+        if not first_test_results_found:
+            continue
+
+        if oldrec[0].isupper():
+            header = oldrec
+
+            if oldrec != newrec:
+                print('File structure has changed')
+                sys.exit()
+
+            continue
+
+        if len(oldvals) != len(newvals):
+            print()
+            print(header[:-1])
+            print(oldrec[:-1])
+            print(newrec[:-1])
+            print('Mismatch in number of numeric values')
+            prev_header = header
+            prev_oldrec = oldrec
+
+        percentages = []
+        discrepancy = False
+        for j in range(min(len(oldvals), len(newvals))):
+            x = float(oldvals[j])
+            y = float(newvals[j])
+
+            if x == 0. and y == 0.:
+                percent = 0.
+            else:
+                percent = 200. * abs((x-y)/(x+y))
+
+            percentages.append(percent)
+            if percent > 0.1:
+                discrepancy = True
+
+        if not percentages:
+            continue
+
+        if verbose or discrepancy:
+            if prev_header != header:
+                print()
+                print(header[:-1])
+                prev_header = header
+
+            suffixes = []
+            for percent in percentages:
+                suffixes.append('%.4f' % percent)
+
+            if discrepancy and verbose:
+                stars = ' ********'
+            else:
+                stars = ''
+
+            print(oldrec[:-1])
+            print(newrec[:-1], '  (' + ', '.join(suffixes) + ')' + stars)
+
+
+
+#*******************************************************************************
+class Backplane_Settings(object):
+
+    from oops.unittester_support            import TESTDATA_PARENT_DIRECTORY
+
+    ARGS = []
+    DIFF = []
+    EXERCISES_ONLY = False
+    NO_EXERCISES = False
+    NO_COMPARE = False
+    SAVING = True
+    LOGGING = False
+    PRINTING = False
+    UNDERSAMPLE = 16
+#    PLANET_KEY = None
+#    MOON_KEY = None
+#    RING_KEY = None
+    OUTPUT = None
+    REFERENCE = None
+    REF = False
+
+
+
+#===============================================================================
+def backplane_unittester_args():
+    """
+    Parse command-line arguments for backplane unit tests.  Results are
+    stored as Backplane_Settings attributes.
+    """
+
+    import argparse
+    import sys
+
+
+    ## Define arguments ##
+    parser = argparse.ArgumentParser(description='Backplane unit tester.')
+
+
+    # Generic arguments
+    parser.add_argument('--args', nargs='*', metavar='arg', default=None,
+                        help='Generic arguments to pass to the test modules. ' \
+                             'Must occur last in the argument list.')
+
+
+    # Basic controls
+    parser.add_argument('--verbose', action='store_true', default=None,
```

Comment:
I think the default for a boolean flag should be False instead of None

Reply:
This is intentional.  I'm using the defaults defined as Backplane_Settings attributes, so  defaulting everything to None here allows me to detect whether or not the argument has been supplied, and therefore whether to change it from the what was set there.  I find this much easier to read and manage.